pkg/cli/admin/catalog/mirror.go

```diff
@@ -485,35 +522,26 @@ func generateICSP(out io.Writer, name string, icspScope string, mapping map[imag
 		},
 	}
 
-	registryMapping := map[string]string{}
-	for k, v := range mapping {
-		if len(v.Ref.ID) == 0 {
-			fmt.Fprintf(out, "no digest mapping available for %s, skip writing to ImageContentSourcePolicy\n", k)
-			continue
-		}
-		if icspScope == "registry" {
-			registryMapping[k.Ref.Registry] = v.Ref.Registry
-		} else {
-			registryMapping[k.Ref.AsRepository().String()] = v.Ref.AsRepository().String()
-		}
-	}
 	for key := range registryMapping {
 		icsp.Spec.RepositoryDigestMirrors = append(icsp.Spec.RepositoryDigestMirrors, operatorv1alpha1.RepositoryDigestMirrors{
 			Source:  key,
 			Mirrors: []string{registryMapping[key]},
 		})
-	}
+		y, err := yaml.Marshal(icsp)
+		if err != nil {
+			return nil, fmt.Errorf("unable to marshal ImageContentSourcePolicy yaml: %v", err)
+		}
 
-	// Create an unstructured object for removing creationTimestamp
-	unstructuredObj := unstructured.Unstructured{}
-	var err error
-	unstructuredObj.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&icsp)
-	if err != nil {
-		return nil, fmt.Errorf("error converting to unstructured: %v", err)
+		if len(y) > byteLimit {
+			if len(icsp.Spec.RepositoryDigestMirrors) > 0 {
+				icsp.Spec.RepositoryDigestMirrors = icsp.Spec.RepositoryDigestMirrors[:len(icsp.Spec.RepositoryDigestMirrors)-1]
```

Comment:
Won't this just limit the number of mappings and then return? Don't we need to generate multiple files for the overflow?

Reply:
> Won't this just limit the number of mappings and then return?
Yes, but this function is called until all entries in the map are removed.

> Don't we need to generate multiple files for the overflow?
The output of this change is a single file with multiple yamls separated by `---`.

Reply:
Example Yaml:
```apiVersion: operator.openshift.io/v1alpha1
kind: ImageContentSourcePolicy
metadata:
  name: redhat-operator-index-0
spec:
  repositoryDigestMirrors:
  - mirrors:
    - quay.io/agreene/new-index-openshift4-ose-sriov-network-operator
    source: registry.redhat.io/openshift4/ose-sriov-network-operator
  - mirrors:
    - quay.io/agreene/new-index-integration-service-registry-infinispan-rhel8
    source: registry.redhat.io/integration/service-registry-infinispan-rhel8
  - mirrors:
    - quay.io/agreene/new-index-openshift-serverless-1-tech-preview-serving-webhook-rhel8
    source: registry.redhat.io/openshift-serverless-1-tech-preview/serving-webhook-rhel8
---
apiVersion: operator.openshift.io/v1alpha1
kind: ImageContentSourcePolicy
metadata:
  name: redhat-operator-index-1
spec:
  repositoryDigestMirrors:
  - mirrors:
    - quay.io/agreene/new-index-distributed-tracing-jaeger-es-rollover-rhel8
    source: registry.redhat.io/distributed-tracing/jaeger-es-rollover-rhel8
  - mirrors:
    - quay.io/agreene/new-index-openshift-pipelines-tech-preview-pipelines-triggers-webhook-rhel8
    source: registry.redhat.io/openshift-pipelines-tech-preview/pipelines-triggers-webhook-rhel8
  - mirrors:
    - quay.io/agreene/new-index-openshift-service-mesh-grafana-rhel8
    source: registry.redhat.io/openshift-service-mesh/grafana-rhel8
---
apiVersion: operator.openshift.io/v1alpha1
kind: ImageContentSourcePolicy
metadata:
  name: redhat-operator-index-2
spec:
  repositoryDigestMirrors:
  - mirrors:
    - quay.io/agreene/new-index-openshift-serverless-1-tech-preview-serving-autoscaler-rhel8
    source: registry.redhat.io/openshift-serverless-1-tech-preview/serving-autoscaler-rhel8
  - mirrors:
    - quay.io/agreene/new-index-openshift-pipelines-tech-preview-pipelines-entrypoint-rhel8
    source: registry.redhat.io/openshift-pipelines-tech-preview/pipelines-entrypoint-rhel8
  - mirrors:
    - quay.io/agreene/new-index-codeready-workspaces-jwtproxy-rhel8
    source: registry.redhat.io/codeready-workspaces/jwtproxy-rhel8```